internal_functions/functions.h

```diff
@@ -0,0 +1,35 @@
+#include <cstdio>
+#include <cstddef>
+#include <string>
+
+
+#ifndef MYSHELL_MYSHELL_FUNCTIONS_H
+#define MYSHELL_MYSHELL_FUNCTIONS_H
+
+// Output error number.
+// supports [-h | --help]
+extern int merrno();
```

Comment:
I prefer way when each command takes as arguments "int arc, char* argv" and decide what to do with them.

Reply:
but then we'll have to repeat argument-checking logic in every function. Instead, we could be handling them in the main part and invoke functions with correct arguments only.